velox/exec/fuzzer/AggregationFuzzerBase.cpp

```diff
@@ -300,16 +300,29 @@ std::vector<RowVectorPtr> AggregationFuzzerBase::generateInputData(
 std::vector<RowVectorPtr> AggregationFuzzerBase::generateInputDataWithRowNumber(
     std::vector<std::string> names,
     std::vector<TypePtr> types,
+    const std::vector<std::string>& partitionKeys,
     const CallableSignature& signature) {
   names.push_back("row_number");
   types.push_back(BIGINT());
 
   auto generator = findInputGenerator(signature);
 
   std::vector<RowVectorPtr> input;
-  auto size = vectorFuzzer_.getOptions().vectorSize;
+  vector_size_t size = vectorFuzzer_.getOptions().vectorSize;
   velox::test::VectorMaker vectorMaker{pool_.get()};
   int64_t rowNumber = 0;
+
+  std::unordered_set<std::string> partitionKeySet;
+  partitionKeySet.reserve(partitionKeys.size());
+  for (auto partitionKey : partitionKeys) {
+    partitionKeySet.insert(partitionKey);
+  }
+  // Set the maximum number of partitions possible to 10 when 10 < size < 100,
+  // and ensure that it increases linearly with size when size > 100. When
+  // size < 10, limit the maximum number of partitions possible to size + 1.
+  vector_size_t maxPartitions =
```

Comment:
Doing so limits us from generating partitions with very few rows now...e.g. we would definitely want to test input where there is only one row per partition.

So generate a random number and sometimes generate maxPartitions > number of rows/3 or 4 so that those cases are generated as well.

Reply:
Updated, thanks for the suggestion @aditi-pandit. Could you please take another look?

Reply:
@pramodsatya : This is still limiting the maxPartitions to be numRows/3... we want cases where numPartitions is between numRows/3 and numRows.

Reply:
Got it, I have updated maxPartitions to be a random integer between `size/3` and `size`, with a lower bound of 1. Would this be fine @aditi-pandit?